File: common/src/main/java/de/werkelmann/rekote/address/url/UrlLabelsLengthRule.java
package de.werkelmann.rekote.address.url;

import de.werkelmann.rekote.address.ConnectionRule;

public class UrlLabelsLengthRule implements ConnectionRule {
    @Override
    public boolean isValid(String address) {
        for (String label : address.split(".")) {
            if (label.length() > 63) {
                return false;
            }
        }
        return true;
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/url/UrlLabelsLengthRule.kt
package de.werkelmann.rekote.address.url

import de.werkelmann.rekote.address.ConnectionRule

class UrlLabelsLengthRule : ConnectionRule {
    override fun isValid(address: String): Boolean {
        address.split(".".toRegex()).forEach {
            if (it.length > 63) {
                return false
            }
        }
        return true
    }
}
